spec/lang/syntax.md

```diff
@@ -110,6 +110,8 @@ pub enum CastOp {
     /// The program is well-formed even if the output type has a different size than the
     /// input type, but the operation is UB in that case.
     Transmute(Type),
+    /// Returns the metadata of a wide pointer as a value.
+    GetMetadata,
```

Comment:
This doesn't feel like a cast to me. It can't be done with `as` or anything similar in Rust.

Please make it an `UnOp`.